src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -251,6 +251,10 @@ static void checkAccessRightsForSelect(
 
     /// General check.
     context->checkAccess(AccessType::SELECT, table_id, syntax_analyzer_result.requiredSourceColumnsForAccessCheck());
+
+    /// Need access SYSTEM_CONFIGS to select system.configs.
+    if (table_id.getFullNameNotQuoted() == "system.configs")
+        context->checkAccess(AccessType::SYSTEM_CONFIGS, table_id);
```

Comment:
Why this cannot be placed into `src/Storages/System/StorageSystemConfigs.cpp`? This will looks more cleaner.